cocos/core/utils/misc.ts

```diff
@@ -233,8 +171,13 @@ export function isPlainEmptyObj_DEV (obj): boolean {
  * var v1 = clampf(20, 0, 20); // 20;
  * var v2 = clampf(-1, 0, 20); //  0;
  * var v3 = clampf(10, 0, 20); // 10;
+ *
+ * @deprecated since v3.9.0, please use `math.clamp` instead.
  */
 export function clampf (value: number, min_inclusive: number, max_inclusive: number): number {
+    if (DEBUG) {
+        warnID(16001, 'misc.clampf', '3.9.0', 'math.clamp');
+    }
```

Comment:
Can just replace the implementation with `math.clamp()`.

Reply:
yes, but I wanna add some warnning here for `misc.clampf`, then we cannot strightfully replace the implementation with `math.clamp()`.

Reply:
You can do it like this
```typescript
export function clampf (value: number, min_inclusive: number, max_inclusive: number): number {
    if (DEBUG) {
        warnID(16001, 'misc.clampf', '3.9.0', 'math.clamp');
    }
   return math.clmap(value, min_inclusive, max_inclusive);
}
```

Reply:
ah, got it, sure we should do it like this, I updated here https://github.com/cocos/cocos-engine/pull/16365/commits/ba73061c9e468ad61bb76b7e93b92975192b6ca6